src/concept/answer/ConceptMap.jl

```diff
@@ -4,12 +4,42 @@ struct ConceptMap
     data::Dict{String, AbstractConcept}
 end
 
-# Java uses "unmodifiableMap" but if that's the case I would probably hide
-# the implementation here and not expose the underlying Dict. Hence commenting
-# out the following function.
-# map(cm::ConceptMap) = cm.data
+struct Explainable
+    conjunction::String
+    explainable_id::Int64
+end
+
+mutable struct Explainables
+    relations::Optional{Dict{String,Explainable}}
+    attributes::Optional{Dict{String,Explainable}}
+    ownerships::Optional{Dict{Tuple{String},Explainable}}
+end
```

Comment:
I haven't looked into other code. Does this need to be mutable?

Reply:
If the fields are Optional, I think so (if that means that the dicts can be added later).
Another way of handling it would be to make it an immutable struct, and initialize it with empty Dicts.